app/models/setting.rb

```diff
@@ -30,16 +30,21 @@ def validate(record)
     end
   end
 
+  def self.allows_blank?(name)
+    name = name.to_s
+    BLANK_ATTRS.include?(name) || (Foreman::SettingManager.settings.key?(name) && Foreman::SettingManager.settings[name][:options][:allow_blank])
+  end
+
   validates_lengths_from_database
 
   validates :name, :presence => true, :uniqueness => true
   validates :description, :presence => true
-  validates :default, :presence => true, :unless => proc { |s| s.settings_type == "boolean" || BLANK_ATTRS.include?(s.name) }
+  validates :default, :presence => true, :unless => proc { |s| s.settings_type == "boolean" || s.class.allows_blank?(s.name) }
   validates :default, :inclusion => {:in => [true, false]}, :if => proc { |s| s.settings_type == "boolean" }
```

Comment:
I just didn't want to break current behaviour, but I'd be in favor of removing the validations for `default`, as it only validates code, not user input.

Reply:
I actually like it as a safeguard for developers. However, that should probably be done when loading the settings rather than as a DB level change. So probably in `SettingManager` instead.

Reply:
It is there.. you need to pass a default :)
There is no type validation on it though